src/ActionList/List.tsx

```diff
@@ -153,10 +147,11 @@ export function List(props: ListProps): JSX.Element {
   let groups: (GroupProps | (Partial<GroupProps> & {renderItem?: typeof Item; renderGroup?: typeof Group}))[] = []
 
   // Collect rendered `Item`s into `Group`s, avoiding excess iteration over the lists of `items` and `groupMetadata`:
+  const singleGroupId = useMemo(uniqueId, [])
```

Comment:
To clarify, this is here to ensure the same number of hooks is called on each render, i.e. to avoid conditionally calling `useMemo` as Group count changes?

Reply:
Yep, that's exactly why.  Hooks shouldn't be called inside loops or conditionals to ensure consistent execution order across renders.